### PR TITLE
cdc: batch send resolved ts exponentially to speed up TiCDC resolve lock

### PR DESCRIPTION
**Pull request by overvenus**
### What is changed and how it works?

Issue Number: ref #11993

Close https://github.com/pingcap/tiflow/issues/8561

What's Changed:

Batch send resolved ts exponentially to speed up TiCDC resolve lock

```commit-message
cdc: batch send resolved ts exponentially to speed up TiCDC resolve lock
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by hicqu — APPROVED** [review]

File: components/cdc/src/endpoint.rs [hunk 1]
```diff
@@ -837,7 +830,7 @@ impl<T: 'static + RaftStoreRouter<E>, E: KvEngine> Endpoint<T, E> {
 
     fn on_min_ts(&mut self, regions: Vec<u64>, min_ts: TimeStamp, current_ts: TimeStamp) {
         // Reset resolved_regions to empty.
-        let resolved_regions = &mut self.resolved_region_heap;
```
Comment (hicqu) [inline components/cdc/src/endpoint.rs:840]: If we use `self.resolved_region_heap` directly, seems we can avoid `RefCell`.

Comment (overvenus) [reply components/cdc/src/endpoint.rs:840]: It won't compile, because there is a reference to self (`broadcast_resolved_ts`) after the mut reference.

**Review by 3pointer — APPROVED** [review]

**Pull request merged**